lightly/models/utils.py

```diff
@@ -596,3 +597,72 @@ def repeat_interleave_batch(x, B, repeat):
         dim=0,
     )
     return x
+
+
+def _get_2d_sincos_pos_embed(embed_dim, grid_size, cls_token=False):
```

Comment:
```suggestion
def get_2d_sincos_pos_embed(embed_dim: int, grid_size: int, cls_token: bool = False) -> NDArray[np.float_]:
```

We use `_function_name` for private functions (only used in the same file) and `function_name` for public functions that can be used in different files. Could you make all the functions public?

I also added type hints, could you add them to the other functions as well?